src/peft/tuners/lora/tp_layer.py

```diff
@@ -0,0 +1,160 @@
+from typing import Any
+
+import torch
+import torch.nn as nn
+import torch.nn.init as init
+
+from .layer import LoraLayer
+
+
+class LoraParalleLayer(LoraLayer):
+    def __init__(self, base_layer, backend, is_paralle_a: bool = False):
+        LoraLayer.__init__(self, base_layer=base_layer)
+        self.backend = backend
+        self.is_paralle_a = is_paralle_a
+
+    def update_layer(self, adapter_name, r, lora_alpha, lora_dropout, init_lora_weights, **parallel_linear_kwargs):
+        if r <= 0:
+            raise ValueError(f"`r` should be a positive integer value but the value passed is {r}")
+        self.r[adapter_name] = r
+        self.lora_alpha[adapter_name] = lora_alpha
+        if lora_dropout > 0.0:
+            lora_dropout_layer = nn.Dropout(p=lora_dropout)
+        else:
+            lora_dropout_layer = nn.Identity()
+
+        self.lora_dropout.update(nn.ModuleDict({adapter_name: lora_dropout_layer}))
+
+        config = parallel_linear_kwargs["config"]
+        # lora needs to be forced to upgrade to 32-bit precision, otherwise it will overflow
+        config.params_dtype = torch.float32
+        init_method = parallel_linear_kwargs.get("init_method", init.xavier_normal_)
+        input_is_parallel = parallel_linear_kwargs.get("input_is_parallel", True)
+        gather_output = parallel_linear_kwargs.get("gather_output", False)
+        if r > 0:
```

Comment:
> This is not needed because we already check `r <= 0` at the beginning, right? Yes, we should also remove the check in the default LoRA layer :)

Ha, ok~~